### DISCUSSION TITLE
Network Analysis in R Studio

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#595] [Q&A] [answered]
Selamat siang kak. Saya punya data seperti ini : 

<img width="503" alt="image" src="https://user-images.githubusercontent.com/106010284/225813193-618afb76-c582-46d0-a3b4-cb10e5a7185b.png">

Saya ingin melihat perjalanan dari "a". Kodenya seperti ini : 
<img width="464" alt="image" src="https://user-images.githubusercontent.com/106010284/225813450-2db4e2b9-4283-4f05-aa0f-cdfc91ec842b.png">

Outputnya seperti ini :  
<img width="331" alt="image" src="https://user-images.githubusercontent.com/106010284/225813504-27524ac7-129f-43da-ac78-8b6b3c5bc328.png">

Bagaimana caranya jika ingin menampilkan sampai derajat kedua (perjalanan "c", "d", dan "z"), derajat ketiga, dst ?

Mohon bantuannya kak, terima kasih 🙏

**Comment by amaliasf** [comment]
Dear @yohanespm77 
Terima kasih atas pertanyaannya pak, tim kami akan segera coba bantu atau carikan referensi ya pak

**Comment by amaliasf** [comment, marked as answer]
Dear @yohanespm77 

Untuk membuat grafik terhadap semua garis yang terhubung berdasarkan kolom "from" dan "to", Bapak bisa menambah nilai lain pada kondisi filter seperti code sebagai berikut:

```
gr2 <- graph_from_data_frame((data_test %>% 
                                filter(from %in% c("a","b","c","d"))), directed = TRUE)
```

Boleh silahkan dicoba Pak, apakah sesuai dengan graf yang Bapak harapkan. Silahkan infokan kembali kalau masih belum sesuai. Semoga membantu Pak.

**Reply by yohanespm77** [reply]
Graphnya sesuai kak. Tapi yang saya harapkan ketika ingin melihat perjalanan "a" dengan mengatur derajat networknya = 1, maka outputnya : 
<img width="319" alt="image" src="https://user-images.githubusercontent.com/106010284/225876464-2679907e-13c0-489e-81de-460cd782d961.png">

Ketika kita mengatur derajat networknya = 2, maka outputnya : 
<img width="385" alt="image" src="https://user-images.githubusercontent.com/106010284/225876794-2c4cf615-06a8-4273-96c7-045a5a1b3852.png">

Ketika kita mengatur derajat networknya = 3, maka outputnya : 
<img width="323" alt="image" src="https://user-images.githubusercontent.com/106010284/225877224-e70b0173-e996-4ec1-9fa4-d695ae3fb594.png">

dan seterusnya. 

Dengan ketentuan titik pusatnya tetap di "a" (yang difilter hanya "a" saja) kak.
